crates/walrus-sdk/src/config/upload_mode.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! Preset upload modes that tune client communication for throughput/latency.
+
+use serde::{Deserialize, Serialize};
+
+use crate::config::communication_config::ClientCommunicationConfig;
+
+/// Upload preset modes for tuning client concurrency and network usage.
+///
+/// These presets only adjust in-memory ClientCommunicationConfig and are intended
+/// to be applied per-run by the CLI (or other callers). They do not mutate the
+/// on-disk configuration.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub enum UploadMode {
+    /// Targets ~10–100 Mbps links with low memory footprint and modest per-node pressure.
+    /// Sets bytes-in-flight near 100 Mb (~12.5 MB) and a conservative per-node cap.
+    Conservative,
+    /// Targets ~1 Gbps links with balanced throughput and stability.
+    /// Sets bytes-in-flight near 1 Gb (~125 MB) and moderate per-node cap.
+    #[default]
+    Balanced,
+    /// Targets ~10 Gbps environments where higher memory and per-node parallelism are acceptable.
+    /// Sets bytes-in-flight near 10 Gb (~1.25 GB) and higher per-node cap.
+    Aggressive,
+}
+
+struct CommunicationParameters {
+    max_concurrent_writes: usize,
+    max_data_in_flight: usize,
+}
+
+impl UploadMode {
+    fn to_communication_parameters(self) -> CommunicationParameters {
+        match self {
+            UploadMode::Conservative => CommunicationParameters {
+                max_concurrent_writes: 1000,
+                max_data_in_flight: 12_500_000, // 12.5 MB
+            },
+            UploadMode::Balanced => CommunicationParameters {
+                max_concurrent_writes: 1000,
+                max_data_in_flight: 125_000_000, // 125 MB
+            },
+            UploadMode::Aggressive => CommunicationParameters {
+                max_concurrent_writes: 1280,
+                max_data_in_flight: 1_250_000_000, // 1.25 GB
+            },
+        }
+    }
+
+    /// Applies this preset to a given communication config and returns the updated config.
+    /// If `force` is true, overrides values unconditionally; otherwise only fills defaults.
+    pub fn apply_to(
+        self,
+        mut config: ClientCommunicationConfig,
+        force: bool,
```

Comment:
AFAICT, this is now always `true`. I think we should either "bring back" the `force` parameter or remove this function parameter.

Reply:
+1